contrib/babelfishpg_common/src/varchar.c

```diff
@@ -1102,6 +1110,82 @@ varchar2numeric(PG_FUNCTION_ARGS)
 	PG_RETURN_NUMERIC(result);
 }
 
+#define FIXEDDECIMAL_MULTIPLIER_NEW 100LL
+#define FIXEDDECIMAL_SCALE_NEW 2
+Datum
+fixeddecimal2varchar(PG_FUNCTION_ARGS)
+{
+	int64		val = PG_GETARG_INT64(0);
+	int32		maxByteLen = PG_GETARG_INT32(1);
+	char		buf[MAXINT8LEN + 1];
+	char	   *end;
+	int32		len;
+	Datum		res;
+
+	/* fetch function pointer for cross-module calls. */
+	if (fixeddecimal2str_p == NULL)
+		fixeddecimal2str_p = (fixeddecimal2str_t)
+			load_external_function("$libdir/babelfishpg_money", "fixeddecimal2str", true, NULL);
+
+	end = fixeddecimal2str_p(val, buf, FIXEDDECIMAL_MULTIPLIER_NEW, FIXEDDECIMAL_SCALE_NEW);
+	len = (end - buf);
+
+	maxByteLen -= VARHDRSZ;
+	if (maxByteLen >= 0 && len > maxByteLen)
+		ereport(ERROR,
+				(errcode(ERRCODE_STRING_DATA_RIGHT_TRUNCATION),
+				 errmsg("There is insufficient result space to convert a money/smallmoney value to varchar.")));
+
+	res = DirectFunctionCall3(varcharin,
+							   CStringGetDatum(buf),
+							   ObjectIdGetDatum(0),
+							   Int32GetDatum(-1));
+
+	PG_RETURN_DATUM(res);
+}
+
+Datum
+fixeddecimal2bpchar(PG_FUNCTION_ARGS)
+{
+	int64		val = PG_GETARG_INT64(0);
+	int32		maxByteLen = PG_GETARG_INT32(1);
+	char		buf[MAXINT8LEN + 1];
+	char	   *buf_padded;
+	char	   *end;
+	int32		len;
+	Datum		res;
+
+	/* fetch function pointer for cross-module calls. */
+	if (fixeddecimal2str_p == NULL)
+		fixeddecimal2str_p = (fixeddecimal2str_t)
+			load_external_function("$libdir/babelfishpg_money", "fixeddecimal2str", true, NULL);
+
+	end = fixeddecimal2str_p(val, buf, FIXEDDECIMAL_MULTIPLIER_NEW, FIXEDDECIMAL_SCALE_NEW);
+	len = (end - buf);
+
+	if (maxByteLen < 0)
+		maxByteLen = len + VARHDRSZ;
+	maxByteLen -= VARHDRSZ;
+	if (len > maxByteLen)
```

Comment:
Why no:
```
if (maxByteLen >= 0 && len > maxByteLen)
```

Reply:
Note that we are checking if (maxByteLen < 0) then we are adjusting it to (len + VARHDRSZ) so it will never be negative at this point. I kept this different from the function for varchar (`fixeddecimal2bpchar`) as there I am not using this maxByteLen but here I am using it to add padding.
But for simplicity, updated the logic to be same at both the places.